unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+
+#pragma warning disable CS0162
+
+namespace DCL.Helpers.NFT.Markets.OpenSea_Internal
+{
+    internal class RequestController
+    {
+        internal const bool VERBOSE = false;
+
+        List<RequestGroup> requestGroup = new List<RequestGroup>();
+
+        public Request AddRequest(string assetContractAddress, string tokenId)
+        {
+            RequestGroup group = null;
+            for (int i = 0; i < requestGroup.Count; i++)
+            {
+                if (requestGroup[i].isOpen)
+                {
+                    group = requestGroup[i];
+                    break;
+                }
+            }
+            if (group == null)
+            {
+                group = CreateNewGroup();
+            }
+            return group.AddRequest(assetContractAddress, tokenId);
+        }
+
+        RequestGroup CreateNewGroup()
+        {
+            if (VERBOSE) Debug.Log("RequestGroup created");
+            RequestGroup group = new RequestGroup(OnGroupClosed);
+            requestGroup.Add(group);
+            return group;
+        }
+
+        void OnGroupClosed(RequestGroup group)
+        {
+            if (VERBOSE) Debug.Log("RequestGroup closed");
+            if (requestGroup.Contains(group))
+            {
+                requestGroup.Remove(group);
+            }
+        }
+    }
+
+    class RequestGroup : IDisposable
+    {
+        const string API_URL_ASSETS = "https://api.opensea.io/api/v1/assets?";
+        const float WAIT_TIME = 0.2f;
+
+        public bool isOpen { private set; get; }
+
+        List<Request> requests = new List<Request>();
+
+        Coroutine fetchRoutine = null;
+        Action<RequestGroup> onGroupClosed = null;
+
+        public RequestGroup(Action<RequestGroup> onGroupClosed)
+        {
+            isOpen = true;
+            this.onGroupClosed = onGroupClosed;
+            fetchRoutine = CoroutineStarter.Start(Fetch());
+        }
+
+        public Request AddRequest(string assetContractAddress, string tokenId)
+        {
+            Request request = new Request(assetContractAddress, tokenId);
+            requests.Add(request);
+            return request;
+        }
+
+        IEnumerator Fetch()
+        {
+            yield return new WaitForSeconds(WAIT_TIME);
+            isOpen = false;
+            this.onGroupClosed?.Invoke(this);
+            this.onGroupClosed = null;
+
+            string url = API_URL_ASSETS;
+
+            for (int i = 0; i < requests.Count; i++)
+            {
+                url += requests[i].ToString();
```

Comment:
There is a maximum number of characters for a single URL, we might need to check this to limit the number of assets added in each group to avoid surpassing it.

Reply:
> There is a maximum number of characters for a single URL, we might need to check this to limit the number of assets added in each group to avoid surpassing it.

Thanks, I will look into that